### PR TITLE
Add the instruction on how to run single rekt test

### LINKED ISSUE #3297: Add the instruction on how to run single rekt test
Sharing some tricks on how to debug rekt based e2e test:
 ```
SYSTEM_NAMESPACE=knative-eventing go test -v -tags=e2e -count=1 -run <The test name that you would like to run>  -parallel=12 -timeout=30m ./test/rekt
```

Take one of your failed test as an example, 
```
 SYSTEM_NAMESPACE=knative-eventing go test -v -tags=e2e -count=1 -run TestEventTransformationForTrigger  -parallel=12 -timeout=30m ./test/e2e_new/
```

This command allows you to run **single** rekt test. You don't have to run all the rekt test and wait for hours for them to be finished.

_Originally posted by @Leo6Leo in https://github.com/knative-extensions/eventing-kafka-broker/issues/3252#issuecomment-1680975774_
            

When you are running kafka broker related test, you may need to specify the broker class

```
 SYSTEM_NAMESPACE=knative-eventing BROKER_CLASS=Kafka go test -v -tags=e2e -count=1 -run TestBrokerTLSCARotation  -parallel=12 -timeout=30m ./test/e2e_new
```

# Description
This ticket is aiming to update DEVELOPMENT.md, to tell the developers what command they can use to run single rekt test. Please also add the explanation for each parameter.

### PR DESCRIPTION
**Pull request by rajibmitra**
Fixes #3297 

**Docs**
This is Docs  PR , added instruction for rekt based test and its parameters.

### CODE REVIEW

File: DEVELOPMENT.md
Comment (Leo6Leo) [inline DEVELOPMENT.md:135]: Put them to a code block might make it look nicer.

**Review by Leo6Leo — CHANGES_REQUESTED** [review]

**Comment by Leo6Leo** [comment]
/retitle Add the instruction on how to run single rekt test

File: DEVELOPMENT.md
Comment (rajibmitra) [reply DEVELOPMENT.md:136]: @Leo6Leo thanks for the review comment, I have added them inside code block, please take a look.

**Comment by Leo6Leo** [comment]
/lgtm
/cc @pierDipi @creydr @Cali0707 
Thanks for the contribution and the prompt action! The instructions and explanation you have added is clear and easy to follow. Great job!

File: DEVELOPMENT.md
Comment (Cali0707) [inline DEVELOPMENT.md:119]:
```suggestion
When you are running kafka broker related tests, you may need to specify the broker class
```

File: DEVELOPMENT.md
Comment (Cali0707) [inline DEVELOPMENT.md:111]: Can you add a note here indicating that they have to replace `<test_name>` with the name of the test?

File: DEVELOPMENT.md
Comment (Cali0707) [inline DEVELOPMENT.md:125]:
```suggestion
Below are descriptions of the parameters you will normally use, as well as some other things you will often need to set while running the tests:
```

File: DEVELOPMENT.md
Comment (Cali0707) [inline DEVELOPMENT.md:129]:
```suggestion
-tags=e2e uses files tagged as e2e when building the tests
```

**Review by Cali0707 — CHANGES_REQUESTED** [review]

**Comment by Leo6Leo** [comment]
Heyy @rajibmitra! Are there anything we could help with on this PR?

**Comment by rajibmitra** [comment]
> Heyy @rajibmitra! Are there anything we could help with on this PR?

Sorry, I didnt get the notification earlier of these comments, I have incorporated all the review comments, please take a look.

**Comment by pierDipi** [comment]
Test failure doesn't seem related 
/retest-required

File: DEVELOPMENT.md
Comment (rajibmitra) [reply DEVELOPMENT.md:111]: done @Cali0707

**Review by Cali0707 — COMMENTED** [review]
/lgtm
/cc @pierDipi

**Comment by rajibmitra** [comment]
is there anything left to be done in this PR, please let me know.

**Comment by Cali0707** [comment]
Just waiting on an eventing approver to approve this change!

**Review by pierDipi — APPROVED** [review]
Thanks!

/approve

**Pull request merged**
